src/bobtail/dag.cpp

```diff
@@ -69,9 +69,52 @@ void CBobtailDag::SetId(int16_t new_id)
     id = new_id;
 }
 
+bool CBobtailDag::CheckForCompatibility(CDagNode* newNode)
+{
+    for (auto &tx : newNode->subblock.vtx)
+    {
+        for (auto &input : tx->vin)
+        {
+            if (spent_outputs.count(input.prevout) != 0)
+            {
+                if (spent_outputs[input.prevout] != tx->GetHash())
+                {
+                    // only add it to incompatible_dags if we are not checking
+                    // if the node can be added to this dag
+                    if (id != newNode->dag_id)
+                    {
+                        incompatible_dags.emplace(newNode->dag_id);
+                    }
+                    return false;
+                }
+            }
+        }
+    }
+    return true;
+}
+
+void CBobtailDag::UpdateCompatibility(const int16_t &new_id, const std::set<int16_t> &old_ids)
```

Comment:
when we merge dags we are taking multiple dags with unique ids and making one new dag. that new dag will use the same id as the dag in the set of merged dags that had the lowest id value. the other ids are removed from vdags because those dags no longer exist. 

This function goes through the incompat list for a given dag and checks for the old dag ids that were now deleted. If we find one of those Ids we insert the new id those dags were merged into because if that new dag is compatible with the olds dags which we were incompatible with, then we must also be incompatible with the new dag.